projects/js-packages/connection/state/actions.jsx

```diff
@@ -87,6 +103,32 @@ function* registerSite( { registrationNonce, redirectUri } ) {
 	}
 }
 
+/**
+ * Side effect action which will fetch a new list of connectedPlugins from the server
+ *
+ * @param {object}   store - Redux store.
+ * @param store.dispatch
+ * @returns {Promise}      - Promise which resolves when connectedPlugins is updated.
+ */
+function requestAndUpdateConnectedPlugins( { dispatch } ) {
+	return new Promise( resolve => {
+		// Fetch.
+		return restApi.fetchConnectedPlugins().then( data => {
+			dispatch( setConnectedPlugins( data ) );
+			resolve( data );
+		} );
+	} );
+}
+
+/**
+ * Side effect action which will fetch a new list of connectedPlugins from the server
+ *
+ * @returns {Promise} - Promise which resolves when the product status is activated.
+ */
+const refreshConnectedPlugins = () => async store => {
+	return await requestAndUpdateConnectedPlugins( store );
```

Comment:
Don't get why split this into two functions if just one place is using it.

Reply:
because of the `await` ?

Reply:
we could use `await` with everything together.

```
return await new Promise( ... )
```

Reply:
https://github.com/Automattic/jetpack/pull/23358/commits/e4a7d556a8ea6a63d7583387b7b6a707d299ba54